### PR TITLE
hot-reload-bot adding hot-reloading config close #1066, close #1100

### LINKED ISSUE #1066: wechaty handling in plain ES6 classes sample
Just a simple noob request,

Can someone provide me with a simple example that handles wechaty in plain ES6 classes please? 

The reason I'm asking is that, 

- I understand that wechaty was initially developed long before ES6 is mature enough to be used, however, 
  * Looking at the [ES6 support in Node.js](http://node.green/), I can see really really few ES6 features are unsupported now. I.e., only very few _extreme_ end-cases are left. 
  * moreover, I don't know how well the transpiling is, e.g., for the advanced ES6 features like `map`, `filter` and `reduce`, I don't know how efficient the transpiled code is.
  * Starting with version 8.5.0, Node.js supports ES modules natively, so we can do `import {add} from './lib.mjs';` directly in ES6 now.
  * I.e., the ES6 is ready for prime time use, writing in TypeScript then have it transpiled into ES6 code seems like a redundant step now. (I'm not saying that TypeScript is not helpful, it is still good in big projects that you have some dummies really easily to shoot their own feet)
- I'm a long time programmer but never did JavaScript programming before, so it'll be a challenge for me to put all these puzzles together, Node.js, ES6 class that handles wechaty the OO way, etc, etc. 
- Just FTA, I'm planning to contribute my code that logs weixin charts as html files back to the wechaty community, so your help is not in vain. I've [long been planning write a tool to do that](https://github.com/suntong/wx-go/tree/master/plugins/logger), however, the platform that I initially chose was so unstable that I have to look for something else, and start a brand new journey to learn and use JavaScript/TypeScript. 

Let me know if you could help, so that I can elaborate more on what I have in mind (but still should be very simple for JavaScript programmers).

Thanks for helping!

### LINKED ISSUE #1100: hot-reload-bot adding hot-reloading config
The [hot-reload-bot demo code](https://github.com/Chatie/wechaty/blob/master/example/hot-reload-bot/index.js) only hot-reload event handling scripts, however, most event handling will need some kind of configuration, so it is nice to hot-reload configurations as well. 

Do you already have some code exist to handle that @Gcaufy? Would you care to update the demo if you do please? 

Otherwise, I'll see if I can do something, _if it is OK_, @Gcaufy, @zixia. But mind you, I've never written any Javascript programs before.

### PR DESCRIPTION
**Pull request by suntong**
the PR should only pull in the `m1` branch, will github pulls in the `emoji` branch as well?

### CODE REVIEW

File: example/monster/README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+Hot import Wechaty listenser functions after change the source code without restart the program
+
+This directory is an example of how to use `hot-import` 
+feature introduced in [this commit](https://github.com/Chatie/wechaty/commit/c47715b4470e7ade9a2590fd3e66985dd7977622). 
+
+The hot-import is based on an npm package [hot-import](https://www.npmjs.com/package/hot-import)
+
+## About Pure ES6 instead of Typescript
+
+Caution! This section is long!
```
Comment (zixia) [inline example/monster/README.md]: This part is really long! ;-)

Firstly, I'd like to thank you for sharing your thoughts with the community!

Secondly, I feel it's worth to write your opinions on TypeScript as a blog post, would you like to post them as a blog post on https://blog.chatie.io/? If so, I think that would be wonderful!

File: example/monster/config.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    development: { host: '127.0.0.1' },
+    production: { host: '192.168.0.11' }
+  },
+  friend_enabled: false,
+}
+//^^^^^ Customization Ends ^^^^^
+//^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+//==========================================================================
+
+module.exports = config
```
Comment (zixia) [inline example/monster/config.js]: how about `export const config = ...` and `export default config`?

File: example/monster/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+/**
+ * Wechaty hot import bot example
+ *
```
Comment (zixia) [inline example/monster/index.js]: Please add your author information at here.

File: example/monster/index.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+async function main() {
+  const config = await hotImport('config.js')
+
+  bot
+    .on('scan',     './listeners/on-scan')
+    .on('login',    './listeners/on-login')
+    .on('message',  './listeners/on-message')
+
+  if (config.friend_enabled)
```
Comment (zixia) [inline example/monster/index.js]: Should be noticed at here that when we are changing `config.js` for `config.friend_enabled`, this block of code will not be run again.

File: example/monster/listeners/on-message.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+const fs = require('fs')
+
+const { MediaMessage } = require('wechaty')
+const { Misc } = require('wechaty/misc')
```
Comment (zixia) [inline example/monster/listeners/on-message.js]: How about to use `import { MediaMessage, Misc } from 'wechaty'` at here?

1. I'm using Node v9.5 now and it seems to support this syntax already.
1. We should import `Misc` from the module standard export.

File: example/monster/run-hot-import-bot.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (zixia) [inline example/monster/run-hot-import-bot.sh]: Please change the script name from `run-hot-import-bot` to `run-monster-bot` for making it more clear for this example directory.

**Comment by suntong** [comment]
k. will fix each of them and get back to you.

About my "ranting" on the Typescript, OK, I'll write a blog instead, and
just keep a link there.

hmm, how to submit a blog? (I've been to the wiki, the faq, etc, but ...)

**Comment by zixia** [comment]
Awesome, thanks for accepting my invitation to write the Chatie Blog!

I had just sent the invitation via the issue at here: https://github.com/Chatie/blog/issues/75

Please fork the blog repository and write your post inside the `_posts` directory, just like others, then send PR back will be ok.

File: example/monster/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# The monster demo
```
Comment (zixia) [inline example/monster/README.md]: Could you please add a **Get Startted** section to the README, descript how to start your demo bot, like run `run-monster-bot.sh`?

**Comment by suntong** [comment]
> Please get at least one approvement from @Chatie/contributor before I could be able to merge it.

@lijiarui / @mukaiu / @JasLin / @xinbenlv -- my PR is at the last stage that need either one of your approval to get merged. Please help. Thx!

**Comment by zixia** [comment]
@suntong Could you please add my wechat? 

My number is 918999, thanks!

**Comment by zixia** [comment]
@suntong Could you please fix this eslint error for your PR?

See details at here: https://travis-ci.org/Chatie/wechaty/jobs/352074835#L2224

Or I can merge it and fix it for you if you do not familiar with `eslint`.

**Comment by suntong** [comment]
Sorry @zixia, I'm not too sure how to fix that `eslint` error, 'cause I was following the instruction from

https://github.com/Chatie/wechaty/pull/1123/files/fff68cb367467b0e69609a3a52e80d642074f8b1#diff-be223886de20eec66d85ba71a3d7c35f

Please merge it then fix it for me. Thx!

**Comment by zixia** [comment]
@suntong Okay it's merged.

And I had just invited you to join the Chatie/Contributor Team, please accept. Thanks!

**Comment by suntong** [comment]
got it. thx!

**Comment by zixia** [comment]
@suntong Awesome, I saw you had joined.

Could you please help yourself to add a bio to https://github.com/Chatie/wechaty/wiki/Contributors? It is a contributor introduction page.

**Pull request merged**